bbhash_iter.go

```diff
@@ -0,0 +1,58 @@
+package bbhash
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"io"
+	"iter"
+
+	"github.com/relab/bbhash/internal/fast"
+)
+
+type BBHashSeq struct {
+	bits       []*bitVector
+	ranks      []uint64
+	reverseMap []uint64
+	hashFunc   int // Use int instead of bool incase there can be more hash functions tested than fasthash64 and SHA256. 0 = hash64, 1 = SHA256.
+	MaxLevel   int
+}
+
+// Find the chunks from slow memory
+// Currently only takes the whole sequence of input and crashes if it's not able to fit everything in the buffer.
+func ReadChunks(readerInfo io.Reader, bufSz int) iter.Seq[[]byte] {
+	return func(yield func([]byte) bool) {
+		i := 0
+		for {
+			// Create buffer and read the input into it for a certain range of bytes
+			buffer := make([]byte, bufSz)
```

Comment:
We can move the buffer allocation out of the for loop and avoid a new allocation on every chunk. This should be okay since we are only reading the chunk into the buffer for the purpose of calculating its hash (the key); we don't need to keep the chunk data around.